### PR TITLE
[DOCS] Reformat "put index template" API docs

### PR DESCRIPTION
**Pull request by jrodewig**
This PR updates the put index template API docs to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to elastic/docs#937 and #43765

### CODE REVIEW

File: docs/reference/indices/templates.asciidoc [hunk 1]
```diff
@@ -1,19 +1,10 @@
...
-based on an index pattern that matches the index name.
-// end::index-template-def[]
-
-NOTE: Templates are only applied at index creation time. Changing a template
-will have no impact on existing indices. When using the create index API, the
-settings/mappings defined as part of the create index call will take precedence
-over any matching settings/mappings defined in the template.
-
-For example:
+Creates or update an index template.
```
Comment (szabosteve) [inline docs/reference/indices/templates.asciidoc]:
```suggestion
Creates or updates an index template.
```

File: docs/reference/indices/templates.asciidoc [hunk 2]
```diff
@@ -42,14 +33,91 @@ PUT _template/template_1
...
+[[put-index-template-api-request]]
+==== {api-request-title}
+
+`PUT /_template/<index-template>`
+
+
+[[put-index-template-api-desc]]
+==== {api-description-title}
+
+Use the put index template API
```
Comment (szabosteve) [inline docs/reference/indices/templates.asciidoc]:
```suggestion
Use the PUT index template API
```

**Review by szabosteve — APPROVED** [review]
I have two small comments to consider, but LGTM.

**Comment by jrodewig** [comment]
038c70b re-adds an anchor that, when removed, breaks a link in the Libbeat docs.

That makes this PR dependent on #46296.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by alisonelizabeth** [comment]
Hi @jrodewig! I noticed this PR removes information about template versioning. Was that intentional?

**Comment by jrodewig** [comment]
Thanks for catching that @alisonelizabeth.

I didn't intend to entirely remove that documentation. I opened a separate PR to re-add it with #46384.

**Comment by alisonelizabeth** [comment]
Thanks @jrodewig!

**Pull request merged**